### ISSUE TITLE
JavadocStyle does not report consistently and not like documented

### ISSUE THREAD

**Issue by Vampire** [checkstyle/checkstyle#6528]
```
/var/tmp $ javac Foo.java
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocStyle"/>
    </module>
</module>
/var/tmp $ cat Foo.java
/**
 * Foo.
 * <foo/>}
 * <foo></foo>}}
 * <foo></foo>}}}
 * <foo>
 * </bar>}}}}
 */
class Foo {
}
/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-8.18-all.jar -c config.xml Foo.java
Starting audit...
[ERROR] D:\Sourcecode\other\discord-logger\tmp\Foo.java:4:9: Extra HTML tag found: </foo>}} [JavadocStyle]
[ERROR] D:\Sourcecode\other\discord-logger\tmp\Foo.java:5:9: Extra HTML tag found: </foo>}}} [JavadocStyle]
[ERROR] D:\Sourcecode\other\discord-logger\tmp\Foo.java:7:4: Extra HTML tag found: </bar>}}}} [JavadocStyle]
Audit done.
Checkstyle ends with 3 errors.
```

---------------

Documentation says:
- start tag without end tag => "Unclosed HTML tag found:"
- end tag without start tag => "Extra HTML tag found:"
- not allowed HTML tag found => unspecified error message

Expected behavior:
- complain 5 times about not allowed HTML tag
- complain 1 time about "Unclosed HTML tag found: <foo>"
- complain 1 time about "Extra HTML tag found: </bar>"
- when complaining about extra HTML tag like is correctly done for `</bar>`, do not include extra characters in the error message

--------------

**Comment by rnveach** [comment]
JavadocStyle doesn't use javadoc grammar for parsing, but instead uses regular expression. Failures in validation could be related to that.

Trying to print the javadoc tree results in this exception:
````
Exception in thread "main" java.lang.IllegalArgumentException: [ERROR:6] Javadoc comment at column 4 has parse error. Missed HTML close tag 'foo'. Sometimes it means that close tag missed for one of previous tags.
    at com.puppycrawl.tools.checkstyle.DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(DetailNodeTreeStringPrinter.java:68)
    at com.puppycrawl.tools.checkstyle.AstTreeStringPrinter.parseAndPrintJavadocTree(AstTreeStringPrinter.java:112)
    at com.puppycrawl.tools.checkstyle.AstTreeStringPrinter.printJavaAndJavadocTree(AstTreeStringPrinter.java:94)
    at com.puppycrawl.tools.checkstyle.AstTreeStringPrinter.printJavaAndJavadocTree(AstTreeStringPrinter.java:98)
    at com.puppycrawl.tools.checkstyle.AstTreeStringPrinter.printJavaAndJavadocTree(AstTreeStringPrinter.java:98)
    at com.puppycrawl.tools.checkstyle.AstTreeStringPrinter.printJavaAndJavadocTree(AstTreeStringPrinter.java:77)
    at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:289)
    at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:180)
    at com.puppycrawl.tools.checkstyle.Main.main(Main.java:119)
````

**Comment by romani** [comment]
This Check do [not use AST](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java#L50), [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L67) is how Checks with AST for javadoc are defined.

JavadocStyle is kind of deprecated. It need to be rewritten from scratch.
BUT when I looked at it https://checkstyle.org/config_javadoc.html#JavadocStyle
its functionality is already mostly covered in other Checks:
> Ensures the first sentence ends with proper punctuation

covered by SummaryJavadoc

> Check text for Javadoc statements that do not have any description.

unclear requirements ... but should be covered by some new Check.

> Check text for incomplete HTML tags. 

this is done by javadoc parser

> Check that a package Javadoc comment is well-formed 

https://github.com/checkstyle/checkstyle/issues/6705 (almost done)

> Check for allowed HTML tags. The list of allowed HTML tags is

Should be done by separate Check, with usage of AST it will be very easy implement.
https://checkstyle.org/writingjavadocchecks.html#HTML_Code_In_Javadoc_Comments

===============

I can not approve this issue, but I am ready to approve New Checks to completely cover JavadocStyle, and deprecate it ones all functionality is covered. We need only "Check for allowed HTML tags" to cover :) all.


If you are interested, please create new issue, feel free to proceed with implementation.
